lib/ci-helper.ts

```diff
@@ -62,6 +65,67 @@ export class CIHelper {
+        // help dugite realize where `git` is...
+        const gitExecutable = os.type() === "Windows_NT" ? "git.exe" : "git";
+        const stripSuffix = `bin${path.sep}${gitExecutable}`;
+        for (const gitPath of (process.env.PATH || "/")
+            .split(path.delimiter)
+            .map((p) => path.normalize(`${p}${path.sep}${gitExecutable}`))
+            // eslint-disable-next-line security/detect-non-literal-fs-filename
+            .filter((p) => p.endsWith(`${path.sep}${stripSuffix}`) && fs.existsSync(p))) {
+            process.env.LOCAL_GIT_DIRECTORY = gitPath.substring(0, gitPath.length - stripSuffix.length);
+            // need to override GIT_EXEC_PATH, so that Dugite can find the `git-remote-https` executable,
+            // see https://github.com/desktop/dugite/blob/v2.7.1/lib/git-environment.ts#L44-L64
+            // Also: We cannot use `await git(["--exec-path"]);` because that would use Dugite, which would
+            // override `GIT_EXEC_PATH` and then `git --exec-path` would report _that_...
+            process.env.GIT_EXEC_PATH = spawnSync("/usr/bin/git", ["--exec-path"]).stdout.toString("utf-8").trimEnd();
+            break;
+        }
+
+        // get the access tokens via the inputs of the GitHub Action
+        this.setAccessToken("gitgitgadget", core.getInput("pr-repo-token"));
+        this.setAccessToken("git", core.getInput("upstream-repo-token"));
+        this.setAccessToken("dscho", core.getInput("test-repo-token"));
+
+        // eslint-disable-next-line security/detect-non-literal-fs-filename
+        if (!fs.existsSync(this.workDir)) await git(["init", "--bare", "--initial-branch", "unused", this.workDir]);
+        for (const [key, value] of [
+            ["gc.auto", "0"],
+            ["remote.origin.url", `https://github.com/${this.config.repo.owner}/${this.config.repo.name}`],
+            ["remote.origin.promisor", "true"],
+            ["remote.origin.partialCloneFilter", "blob:none"],
+        ]) {
+            await git(["config", key, value], { workDir: this.workDir });
+        }
+        console.time("fetch Git notes");
+        await git(
+            [
+                "-c",
+                "remote.origin.promisor=false", // let's fetch the notes with all of their blobs
+                "fetch",
+                "--no-tags",
+                "origin",
+                "--depth=1",
+                `+${GitNotes.defaultNotesRef}:${GitNotes.defaultNotesRef}`,
+            ],
+            {
+                workDir: this.workDir,
+            },
+        );
+        console.timeEnd("fetch Git notes");
+        this.gggNotesUpdated = true;
+        // "Unshallow" the refs by fetching the shallow commits with a tree-less filter.
+        // This is needed because Git will otherwise fall over left and right when trying
+        // to determine merge bases with really old branches.
+        const unshallow = async (workDir: string) => {
```

Comment:
Was there more happening in `unshallow` during development?  Trying to understand why it is not in-line.